pkg/apis/pipeline/v1alpha1/image_resource.go

```diff
@@ -61,11 +61,6 @@ func (s ImageResource) GetType() PipelineResourceType {
 	return PipelineResourceTypeImage
 }
 
-// GetVersion returns the version of the resource
-func (s ImageResource) GetVersion() string {
```

Comment:
same for image, we still need to know which digest to pull, or where is that information coming from? I can't find it

Reply:
image has a `digest` field:

https://github.com/knative/build-pipeline/blob/0c9529db60cc053f9bcd14fc89a8e1454b873746/pkg/apis/pipeline/v1alpha1/image_resource.go#L51

It does mean the logic to interact with these types needs to know what type it's dealing with

(I'm thinking that we could have something kind of cool if in #238 we go the route of using Resources as an extension point, and use duck-typing for resources, then a Resource's controller can take care of itself, so it can know about the Resource's functionality specific attributes like an image's digest or a git resource's commitish)

Reply:
hmm... that sounds interesting

Reply:
We need to also know the type to handle passing these resources for further tasks. It is tough to come up universal solution for resource types.